Multisig/UI/Transactions/TransactionDetails/TransactionDetailsView.swift

```diff
@@ -8,23 +8,74 @@
 
 import SwiftUI
 
-struct TransactionDetailsView: Loadable {
-    @ObservedObject
-    var model: TransactionDetailsViewModel
-    
+struct TransactionDetailsView: View {
+    @FetchRequest(fetchRequest: AppSettings.fetchRequest().all())
+    private var appSettings: FetchedResults<AppSettings>
+
     @FetchRequest(fetchRequest: Safe.fetchRequest().selected())
-    var selectedSafe: FetchedResults<Safe>
+    private var selectedSafe: FetchedResults<Safe>
 
-    var transactionDetails: TransactionViewModel {
-        model.transactionDetails
+    private var safe: Safe { selectedSafe.first! }
+
+    private var canSign: Bool {
+        model.transactionDetails.status == .awaitingConfirmations &&
+            signingKeyAddress != nil &&
+            model.transactionDetails.signers!.contains(signingKeyAddress!) &&
```

Comment:
as signers&confirmations are optionals, I prefer to check if they have values rather than force unwrap

Reply:
It should come here only if the first check pass. If it crashes here then we have a backend issue, that would be a developer error. Crashing, in this case, would be a proper thing to do imo.